File: cakk-domain/mysql/src/main/java/com/cakk/domain/mysql/annotation/DomainFacade.java
package com.cakk.domain.mysql.annotation;

import java.lang.annotation.Documented;
import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import org.springframework.core.annotation.AliasFor;
import org.springframework.stereotype.Component;

/**
 * Indicates that an annotated class is a "Service" (e.g. a domain service object).
 *
 * <p>This annotation serves as a specialization of {@link Component @Component},
 * allowing for implementation classes to be autodetected through classpath scanning.
 *
 * @author komment
 * @see Component
 */

@Target(ElementType.TYPE)
@Retention(RetentionPolicy.RUNTIME)
@Documented
@Component
public @interface DomainFacade {

	/**
	 * Alias for {@link Component#value}.
	 */
	@AliasFor(annotation = Component.class)
	String value() default "";
}

File: cakk-core/src/main/kotlin/com/cakk/core/annotation/DomainFacade.kt
package com.cakk.core.annotation

import org.springframework.core.annotation.AliasFor
import org.springframework.stereotype.Component

/**
 * Indicates that an annotated class is a "Service" (e.g. a domain service object).
 *
 *
 * This annotation serves as a specialization of [@Component][Component],
 * allowing for implementation classes to be autodetected through classpath scanning.
 *
 * @author komment
 * @see Component
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.RUNTIME)
@MustBeDocumented
@Component
annotation class DomainFacade(

	/**
	 * Alias for [Component.value].
	 */
	@get:AliasFor(annotation = Component::class) val value: String = ""
)
